saffron/src/main.rs

```diff
@@ -140,6 +140,56 @@ pub fn verify_storage_proof(args: cli::VerifyStorageProofArgs) -> Result<()> {
     Ok(())
 }
 
+pub fn calculate_diff(args: cli::CalculateDiffArgs) -> Result<()> {
+    let (_, domain) = get_srs(args.srs_cache);
+    let old_data = {
+        let mut file = File::open(args.old)?;
+        let mut buf = Vec::new();
+        file.read_to_end(&mut buf)?;
+        buf
+    };
+    let new_data = {
+        let mut file = File::open(args.new)?;
+        let mut buf = Vec::new();
+        file.read_to_end(&mut buf)?;
+        buf
+    };
+    let diff = make_diff(&domain, &old_data, &new_data);
+    let mut writer = File::create(args.output)?;
+    rmp_serde::encode::write(&mut writer, &diff)?;
+    Ok(())
+}
+
+pub fn update(args: cli::UpdateArgs) -> Result<()> {
+    let (srs, domain) = get_srs(args.srs_cache);
+    let mut blob: FieldBlob<Vesta> = {
+        let file = File::open(args.input.clone())?;
+        rmp_serde::decode::from_read(file)
+    }?;
+    let diff: Diff<Fp> = {
+        let file = File::open(args.diff_file)?;
+        rmp_serde::decode::from_read(file)
+    }?;
+    blob.update::<DefaultFqSponge<VestaParameters, PlonkSpongeConstantsKimchi>>(
+        &srs, &domain, diff,
+    );
+    args.assert_commitment
+        .into_iter()
```

Comment:
The type is `Option<T>`. I won't use a `into_ter()`, even if it works.

Reply:
Yeah this is dumb, it's a haskell pattern but I forgot about `if let` in rust